astrokat/observe_main.py

```diff
@@ -122,13 +122,48 @@ def read_targets(target_items):
     return target_list
 
 
+def slew(session, target_info):
+    """Simple way to get telescope to slew to target
+
+    Parameters
+    ----------
+    session: `CaptureSession` object
+    target_info: dictionary with target observation info
+
+    """
+    katpt_tgt = target_info["target"]
+
+    user_logger.info("Slewing to first target")
+    session.set_target(katpt_tgt)
+    session.activity('slew')
+    # Start moving each antenna to the target
+    if session.kat.dry_run:
+        # Apply average slew time
+        session._slew_to(katpt_tgt)
+    else:
+        # Start moving each antenna to the target
+        session.ants.req.mode('POINT')
+
+    # Initiate FBFUSE proxy settings from mkat-sessions capture_start
+    if session.fbf:
+        user_logger.info('Initiating FBF beam complete')
+        if not session.fbf.check_provisioning_beams_complete():
```

Comment:
@rubyvanrooyen  This check for beam provisioning is potentially a relatively long running blocking function. Does it not perhaps interfere with the timing that MeerTime wants after the target is set, or do they not care about this when slewing to the first target? I seem to remember their commands to trigger their backend being baked into their scripts somehow.

Reply:
This should never had to come up for AstroKAT and should have been fully handled between `mkat_sessions` and the proxies, but there seems to be missing specifications along with a priority need to have this functioning for observations.

I honestly do not agree with the `mkat_sessions` implementation that simply checks if the proxy is present since FBF can run in _commensal mode_ also :(
The presence of the proxy will trigger a wait in `capture_start`, while both `capture_start` and `on_target` seem to rely on this proxy to function if it is included in the array and this worries me.

Discussions with Ewan and Maciej highlighted that the FBFUSE proxy usage has a range or needs.
Since I am not part of the FBFUSE group I do not know the operational needs, but I do understand that they differ from the other USEs in that they under certain situations have the rights to "pause" MKAT observations in order to set up the beams (our current situation).
The implementation as it stands in `mkat_sessions` will impact **ALL** observations that include the FBFUSE proxy and use `capture_start`, even if FBFUSE is running as commensal.

This AstroKAT update simply tries to accommodate the new operational need for the FBF implementation, as well as try to mitigate the impact on observations by "hiding" the potential wait time in the dead time during the first slew.
Else, it will be an impact on the first target observation, which is definitely undesirable.
And the inclusion of the proxy as commensal may be affected, which will mean array build considerations for other observations that do not have this agreement with FBF.

For the moment, a short term solution must be found in order for observations to continue.
During our integration tests all the beam setups was < 90 seconds, which I believe to be acceptable enough to include in the long slew.
If you happen to be on target you will loose a little time, but generally the slew to the first target is your longest dead time during an observation and you do not care about that data, which is why AstroKAT only starts capture when on first target.

The longer term solution is up to the astronomers.
Perhaps @mserylak can advise us here.

Reply:
Hi @rubyvanrooyen and all.
Thank you for the explanation. It is exactly as you mentioned. `mkat_sessions` does not take into account FBFUSE being commensal, thus having no right to "block" the observation. This is at present the only way of circumventing this "feature" in `mkat_sessions` that was discussed between Ruby, Ewan and myself. As far as I am aware, FBFUSE does not require precision timing data for their observations.

As for FBFUSE being commensal with MeerTIME, AstroKAT is not used for these type of observations, but scripts under `katusescripts/ptuse`. And since that type of observations were performed for a while now, I do not think we need to worry. Otherwise it would be signalled to us by MeerTIME people already.

Reply:
@mserylak with regards to the last paragraph, is the intention to no longer use the PTUSE scripts and use astrokat instead thus initiating this PR? I'm not sure how else you would encounter this issue unless there are other observations that need to do things involving target points before the capture_start is requested?

Reply:
@lanceWilliams I guess that would be desired. However, I am not in position to tell if/when this is going to happen. If that is the direction agreed/desired by CAM and PTUSE then I imagine some serious work will need to be done both on AstroKAT and sessions. As you said, `katusescripts/ptuse` has certain commands baked in and I do not know much about that at the momen.